### ISSUE TITLE
Redirect uri overwritten

### ISSUE THREAD

**Issue by jongacnik** [johannschopplich/kirby-locked-pages#1]
Small bug is causing the redirect url to be overwritten, so user is not redirected to the correct page.

**Expected Behavior:**
User should be redirected to `/locked?redirect=page-uri`

**Current Behavior:**
User is redirected to `/locked?redirect=locked`

**Bug:**
`$path` parameter is overwritten. You can either rename the variable, or simply move line 16 after the options array definition.
https://github.com/johannschopplich/kirby-locked-pages/blob/695e496db3b12dae43ce7edcf97921a861d0a2c3/index.php#L16-L20

**Comment by johannschopplich** [comment]
Good catch! Quite a silly mistake... Thank you for noticing as well as the well-structured issue although no issue template was provided in the first place. 🙂

**Comment by Djules** [comment]
New issue with the [1.02 release](https://github.com/johannschopplich/kirby-locked-pages/releases/tag/v1.0.2), the `$path` remains unchanged so there is no redirection actually 😕

The correct approach could be:
```php
$options = [
    'query' => ['redirect' => $path]
];
$slug = option('kirby-extended.locked-pages.slug', 'locked');
go(url($slug, $options));
```

Another issue for me is this line https://github.com/johannschopplich/kirby-locked-pages/blob/101bba460450f7302ee523c40691a9cce66c85e8/index.php#L13

Using Kirby 3.5.0, the `$result` is never empty, so it never goes any further…
A correct test could be:
```php
if (!($result instanceof Page)) return;
```

By the ay, thank your for your plugin! 🙂

**Comment by johannschopplich** [comment]
Gosh, I really should write tests for the plugin. Thanks for both issues! I shouldn't have rushed the "fix" this morning without further testing.

Good to know. I didn't grasp the `$result` object entirely earlier. It seems to be empty only on non-existing routes, like before the error page is thrown. Based on your idea I went with:
```php
if (!is_a($result, Page::class)) return;
```

I've just published a new version, please retest and let me know if it's still broken. Glad to hear you're using my plugin. 🙂

**Comment by Djules** [comment]
@johannschopplich That was quick! 

Just tested 1.0.3 release and it's all good to me, thanks! 👍
